Jenkins_jobs/CreateNewNode.groovy

```diff
@@ -0,0 +1,121 @@
+import jenkins.model.Jenkins;
+import hudson.model.Computer;
+import org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval;
+import hudson.model.Slave;
+import hudson.slaves.JNLPLauncher;
+
+clones = [:]
+List<String> newMachines = new ArrayList<String>();
+
+node {
+    stage('jenkins made me do it') {
+        def nodeHelper = new NodeHelper();
+
+        String[] machines = params.machineNames.split(",")
+        String[] machineIPs = params.machineIPs.split(",")
+        String[] labels = params.labelStrings.split(",")
+
+        def launcher
+        String remoteFS
+        String newMachineLabels
+        String os
+        String newMachineName
+
+        for (int index = 0; index < machineIPs.length; index++) {
+
+            if (nodeHelper.getComputer(machines[index]) == null) {
+                if (Integer.parseInt(machineIPs[index].split("\\.")[0]) == 10) {
+                    launcher = new CommandLauncher(Constants.SSH_COMMAND + "${machineIPs[index]} " + Constants.WGET_SLAVE_JAR);
+                    remoteFS = Constants.REMOTE_FS;
+                } else if (machines[index].contains("win")) {
+                    launcher = new JNLPLauncher("", "", new jenkins.slaves.RemotingWorkDirSettings(false, "", "remoting", false));
+                    remoteFS  = Constants.WIN_REMOTE_FS;
+                } else {
+                    launcher = new SSHLauncher(
+                                machines[index],
+                                22,
+                                params.SSHCredentialId.isEmpty() ? Constants.SSH_CREDENTIAL_ID : params.SSHCredentialId,
+                                null, null, null, null, null, null, null,
+                                new NonVerifyingKeyVerificationStrategy());
+                    remoteFS = Constants.REMOTE_FS;
+                }
+                
+                newMachineLabels = labels[index%labels.length]
+
+                newMachineName = nodeHelper.addNewNode(
+                        machines[index],
+                        machineIPs[index],
+                        remoteFS,
+                        1, // Number of executers 
+                        Mode.EXCLUSIVE,
+                        newMachineLabels.toLowerCase(),
+                        launcher
+                        );
+
+                // This part is to approve the script used to add a 10. machine
+                def scripts  = ScriptApproval.get()
+                def scriptSet = scripts.getPendingScripts()
+                def iterator = scriptSet.iterator()
+                if (launcher.getClass().toString().contains("slaves.CommandLauncher")) {
+                    for (ScriptApproval.PendingScript script : scriptSet) {
+                        // TODO: Search for entire string instead of a part
+                        if (script.script.contains("ssh -C -i /data/jenkins/J9_Build/home/.ssh/j9build.key j9build@" + machineIPs[index])) {
```

Comment:
I don't like this string hardcoded here, shouldn't we be using the Constants file?